core/src/main/scala/kafka/log/LogValidator.scala

```diff
@@ -414,7 +414,9 @@ private[log] object LogValidator extends Logging {
 
       try {
         val recordErrors = new ArrayBuffer[ApiRecordError](0)
-        for ((record, batchIndex) <- batch.asScala.view.zipWithIndex) {
+        var batchIndex = 0
+        while (recordsIterator.hasNext) {
```

Comment:
Are we using the while loop to try and avoid the tuple allocation or something? Seems like that would be noise compared to the other allocations, no?

Reply:
recordsIterator is not an iterable like batches is, so I couldn't figure out how to zip it. I should change this to a scala for at very least.

Reply:
Yeah, `asScala.zip...`. Anyway, your update seems fine.

Reply:
huh, I was trying that but intellij wasn't autocompleting. Not sure what happened.